### PR TITLE
Cache document.currentScript for non-ESM output

### PR DESCRIPTION
**Pull request by CrimsonCodes0**
Fixes issue https://github.com/rustwasm/wasm-bindgen/issues/2502.

Currently, when using non-ESM module output (i.e. scripts output), the generated JavaScript accesses the `Document#currentScript` getter at runtime, which will always have a different (incorrect) value by the time that the user runs their code to load their Wasm.

With this change, the default source for ESM and web script output is cached in an immutable variable, and accessed when the user doesn't pass an argument.

Other output targets should remain untouched.

As an added side-effect, the default source is always constructed, this should add a small amount of extra overhead to loading the glue JS, but if the user loads the Wasm multiple times (for some reason?), it is only constructed once, which should improve performance.

### CODE REVIEW

**Comment by CrimsonCodes0** [comment]
Oh wait, Deno and Node don't get default values, I'll just use the constant variable.

**Comment by CrimsonCodes0** [comment]
Oh wait, Deno and Node don't get default values, I'll just use the constant variable.

**Comment by alexcrichton** [comment]
Thanks! Could you leave a comment in the source for why it's done this way?

**Comment by CrimsonCodes0** [comment]
@alexcrichton Two questions: by "source," are you referring to the JS or Rust? And what about the way that it's done would you like me to leave a comment about?

**Comment by alexcrichton** [comment]
This comment looks fine, thanks for that! I think this needs a rebase now after a previous merge, want to do that and I can merge?

**Comment by CrimsonCodes0** [comment]
Sure, I'll rebase and deal with the conflicts soon.

File: crates/cli-support/src/js/mod.rs [hunk 1]
```diff
@@ -640,35 +640,43 @@ impl<'a> Context<'a> {
...
+                EMPTY
             }
-        } else {
-            String::from("")
         };
 
         let ts = self.ts_for_init_fn(
             has_memory,
-            !self.config.omit_default_module_path && !default_module_path.is_empty(),
+            !self.config.omit_default_module_path && !default_module_path_creator.is_empty(),
```
Comment (CrimsonCodes0) [inline crates/cli-support/src/js/mod.rs:679]: If I'm following this logic correctly, the entire expression could be shortened to the following:
```rust
!default_module_path_creator.is_empty()
```

cc @diceride

If someone omits the default path, and chooses, say, Node.js, there still won't be a default path, therefore we can just check if there is a default path via `.is_empty()`.

**Comment by CrimsonCodes0** [comment]
@alexcrichton It seems like a bunch of runtime tests are now failing after rebasing from the recent merges.

Could you take a look into it and help me debug this?

**Comment by alexcrichton** [comment]
I think you need to run the tests locally with `BLESS=1` to regenerate the expected output since I believe it's changing here

**Comment by alexcrichton** [comment]
(or the test expectations of what the output is just need updating)

**Comment by CrimsonCodes0** [comment]
More recently, I've been responding to people and writing code on mobile, I don't have a PC to download rustc & friends on at the moment, so it would be some time before I could attempt to run those tests locally.

If someone else could run them, we should be able to fix the test failure sooner

**Pull request closed without merging**